### PR TITLE
Remove DataChannelsFactory from constructor

### PR DESCRIPTION
**Pull request by HeyAlex**

### CODE REVIEW

File: app/src/main/java/ru/mail/march/sample/SampleInteractor.kt [hunk 1]
```diff
@@ -1,9 +1,8 @@
 package ru.mail.march.sample
 
-import ru.mail.march.channel.DataChannelFactory
 import ru.mail.march.interactor.Interactor
 
-class SampleInteractor(channelFactory: DataChannelFactory): Interactor, Timer.Observer {
+class SampleInteractor: Interactor(), Timer.Observer {
```
Comment (kdravolin) [inline app/src/main/java/ru/mail/march/sample/SampleInteractor.kt]: https://kotlinlang.org/docs/reference/coding-conventions.html#colon

File: core/src/main/java/ru/mail/march/channel/DataChannelStub.kt [hunk 2]
```diff
@@ -0,0 +1,19 @@
+package ru.mail.march.channel
+
+internal class DataChannelStub<T> : DataChannel<T> {
```
Comment (kdravolin) [inline core/src/main/java/ru/mail/march/channel/DataChannelStub.kt]: StubDataChannel

File: core/src/main/java/ru/mail/march/channel/DataChannelWrapper.kt [hunk 3]
```diff
@@ -0,0 +1,21 @@
+package ru.mail.march.channel
+
+internal open class DataChannelWrapper<T> : DataChannel<T> {
```
Comment (kdravolin) [inline core/src/main/java/ru/mail/march/channel/DataChannelWrapper.kt]: LazyDataChannel

File: core/src/main/java/ru/mail/march/channel/EventChannel.kt [hunk 4]
```diff
@@ -0,0 +1,8 @@
+package ru.mail.march.channel
+
+internal class EventChannel<T> : DataChannelWrapper<T>(),
```
Comment (kdravolin) [inline core/src/main/java/ru/mail/march/channel/EventChannel.kt]: LazyEventChannel

File: core/src/main/java/ru/mail/march/channel/Initializer.kt [hunk 5]
```diff
@@ -0,0 +1,5 @@
+package ru.mail.march.channel
+
+internal interface Initializer<T> {
+    fun init(factory: DataChannelFactory)
```
Comment (kdravolin) [inline core/src/main/java/ru/mail/march/channel/Initializer.kt]: Maybe just add abstract method `init` to LazyDataChannel and remove this interface?

File: app/src/main/java/ru/mail/march/sample/SampleInteractor.kt (hunk 1 above)
Comment (HeyAlex) [reply app/src/main/java/ru/mail/march/sample/SampleInteractor.kt]: need to add ktlint

**Review by kdravolin — APPROVED** [review]
👍

**Pull request merged**